File: Java/src/main/java/QuestionBank/Solution801_900/Solution806.java
package QuestionBank.Solution801_900;

/**
 * @author windows
 */
public class Solution806 {
    public int[] numberOfLines(int[] widths, String s) {
        int count = 1;
        int sum = 0;
        for (int i = 0; i < s.length(); i++) {
            if (sum + widths[s.charAt(i) - 'a'] > 100){
                count++;
                sum = widths[s.charAt(i) - 'a'];
            }else {
                sum += widths[s.charAt(i) - 'a'];
            }
        }
        return new int[]{count, sum};
    }


    public static void main(String[] args) {
        Solution806 solution806 = new Solution806();
    }
}

File: C++/src/Solution801_900/Solution806.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution806.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> numberOfLines(vector<int>& widths, string s) {
  int line = 1;
  int current = 0;
  for (const auto &item : s){
    int num = current + widths[item - 'a'];
    if (num > 100){
      current = widths[item - 'a'];
      line++;
    }else{
      current = num;
    }
  }
  return {line, current};
}
